thread/thread.cpp

```diff
@@ -928,9 +928,9 @@ R"(
         size_t randomizer = (rand() % 32) * (1024 + 8);
         stack_size = align_up(randomizer + stack_size + sizeof(thread), PAGE_SIZE);
         char* ptr = (char*)photon_thread_alloc(stack_size);
-        auto p = ptr + stack_size - sizeof(thread) - randomizer;
-        (uint64_t&)p &= ~63;
```

Comment:
Pointer arithmetics are straight-forward. No need to change them.

Reply:
error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]